test/test_outcomes.py

```diff
@@ -27,8 +28,29 @@ def test_plurality_should_resolve_ties(self):
     """
 
     def test_plurality_should_returns_general_expected_outcome(self):
-        expected_outcome = [("B", 3)]
+        expected_outcome = Result({"C": 2, "B": 3, "A": 0, "D": 0})
         outcome = o.plurality_outcome(self.s)
 
-        self.assertCountEqual(outcome, expected_outcome)
-        self.assertListEqual(outcome, expected_outcome)
+        self.assertEqual(outcome.winner, expected_outcome.winner)
+        self.assertDictEqual(outcome, expected_outcome)
+
+    def test_for_two_should_returns_general_expected_outcome(self):
+        expected_outcome = Result({"C": 2, "B": 3, "A": 2, "D": 3})
+        outcome = o.for_two_outcome(self.s)
```

Comment:
So this works at run time because it'll automatically call the getWinner function of Result? It is giving compile error

Reply:
Can you please expand on what "compile" error it is giving? I am not currently deploying on docker but all the tests pass without any issues when tested with "pre-commit run --all-files"

Reply:
You can access the winner of a Result both by using the method Result.getWinner() and directly accessing the attribute Result.winner initialized during the creation of the Result obj.
I might implement later a way to automatically update the winner attribute in case the Result obj gets modified (which it shouldn't), or just make the Result obj immutable, or just get rid of the winner attribute and make it only accessible using .getWinner()

Reply:
yeah I think it probably works fine at run time, but I like your idea of just having it be accessible through .getWinner(), that way when the type has multiple results and it picks one winner you won't get compile issue

Reply:
We can do that but it's not gonna resolve the issue that you are having. First of all is not a compile error (also python is not compiled), it is not even an error, it is just a warning of Pylint. I usually prefer using Pylance as it is less verbose. We can fix this warning by casting explicitly every type but it is not really needed for our development purpose. 

About using only getWinner(), I am fine implementing it but if we wanna respect Python conventions and good practices, the other option I didn't think about earlier is to rename getWinner() in winner() and use the @attribute decorator removing the attribute winner, which is by far the most elegant and pythonic way of doing things imho. But at this point, we are mainly talking about syntactic sugar, if you wanna treat python as java we can do that, but there are better ways